### PR TITLE
move profiler.step from training_step_and_backward to optimizer_step_…

### PR DESCRIPTION
**Pull request by gaoteng-git**
https://github.com/PyTorchLightning/pytorch-lightning/issues/8222

Before this fix:
image

After this fix:
image

### CODE REVIEW

File: tests/profiler/test_profiler.py [hunk 1]
```diff
@@ -331,7 +331,7 @@ def test_pytorch_profiler_trainer_ddp(tmpdir, pytorch_profiler):
         files = [file for file in files if file.endswith('.json')]
         assert len(files) == 2, files
         local_rank = trainer.local_rank
-        assert any(f'{local_rank}-training_step_and_backward' in f for f in files)
+        assert any(f'{local_rank}-optimizer_step_and_closure_' in f for f in files)
```
Comment (tchaton) [inline tests/profiler/test_profiler.py:334]: Any idea why it is `optimizer_step_and_closure_` and not `optimizer_step_and_closure` ?

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/profiler/test_profiler.py (hunk 1 above)
Comment (gaoteng-git) [reply tests/profiler/test_profiler.py:334]: Well, both are fine because CI has passed.

**Review by justusschock — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

File: pytorch_lightning/profiler/pytorch.py [hunk 2]
```diff
@@ -195,12 +195,13 @@ class PyTorchProfiler(BaseProfiler):
         "test_step",
         "predict_step",
     }
+    RECORD_FUNCTION_PREFIXE = "optimizer_step_and_closure_"
```
Comment (awaelchli) [inline pytorch_lightning/profiler/pytorch.py:198]: typo in prefix here?

Comment (gaoteng-git) [reply pytorch_lightning/profiler/pytorch.py:198]: Thanks! Fixed here: https://github.com/PyTorchLightning/pytorch-lightning/pull/8308

Comment (awaelchli) [reply pytorch_lightning/profiler/pytorch.py:198]: thanks for following up on it so quickly! cheers

**Pull request merged**